src/SelfServe/SqlX/SqlX.rp.ts

```diff
@@ -30,39 +32,66 @@ export const updateDedicatedGatewayResource = async (sku: string, instances: num
     properties: {
       instanceSize: sku,
       instanceCount: instances,
-      serviceType: "Sqlx",
+      serviceType: "SqlDedicatedGateway",
     },
   };
-  const armRequestResult = await armRequestWithoutPolling({
-    host: configContext.ARM_ENDPOINT,
-    path,
-    method: "PUT",
-    apiVersion,
-    body,
-  });
-  return armRequestResult.operationStatusUrl;
+  const telemetryData = { ...body, selfServeClassName: SqlX.name };
+  const updateTimeStamp = selfServeTraceStart(telemetryData);
+  let armRequestResult;
+  try {
+    armRequestResult = await armRequestWithoutPolling({
+      host: configContext.ARM_ENDPOINT,
+      path,
+      method: "PUT",
+      apiVersion,
+      body,
+    });
+    selfServeTraceSuccess(telemetryData, updateTimeStamp);
+  } catch (e) {
+    const failureTelemetry = { ...body, e, selfServeClassName: SqlX.name };
+    selfServeTraceFailure(failureTelemetry, updateTimeStamp);
+  }
+  return armRequestResult?.operationStatusUrl;
 };
 
 export const deleteDedicatedGatewayResource = async (): Promise<string> => {
   const path = getPath(userContext.subscriptionId, userContext.resourceGroup, userContext.databaseAccount.name);
-  const armRequestResult = await armRequestWithoutPolling({
-    host: configContext.ARM_ENDPOINT,
-    path,
-    method: "DELETE",
-    apiVersion,
-  });
-  return armRequestResult.operationStatusUrl;
+  const telemetryData = { selfServeClassName: SqlX.name };
```

Comment:
add data to differentiate between DELETE and GET Rest calls

Reply:
do you also want to add any account name?
@srinathnarayanan  - I beleieve we ll get user PUID, please confirm if the telemetry is wired that way.

Reply:
We get these params for free on every telemetry message from DE
    databaseAccountName,
    defaultExperience,
    authType,
    subscriptionId,
    platform,
    env